### PR TITLE
Add Github Action badge and remove duplicated CI.

### LINKED ISSUE #540: MIgrate CI tests to use GitHub Actions
Migrate CI tests to GitHub Actions. 
Document in this issue any blockers.

### PR DESCRIPTION
**Pull request by wyk9787**
Fixes #540 

10.3 is the oldest supported Xcode version. Support for any previous versions of Xcode is dropped. macOS 10.15 is the only supported macOS version now.

* Added Github Action badge.

* Removed AppVeyor and its badge.

* Removed most of the Travis CI script with the exception of Golang master version.

References of macOS and Xcode support:

* Github Action currently only supports macOS 10.15.

* Github Action's Xcode support: https://github.com/actions/virtual-environments/blob/master/images/macos/macos-10.15-Readme.md#xcode

* Travis CI's macOS & Xcode support: https://docs.travis-ci.com/user/reference/osx/#macos-version

### CODE REVIEW

**Comment by wyk9787** [comment]
The previous PR's Github Action passed: https://github.com/google/pprof/actions/runs/148062605.

Github Action's script takes effect as expected in this PR. It also includes the version (i.e. a few more XCode versions) that I added in this PR.

@aalexand Could you remove AppVeyor as required check for PR submission in the setting?

**Comment by kalyanac** [comment]
> Could you remove AppVeyor as required check for PR submission in the setting?

Disabled AppVeyor webhook.

**Comment by wyk9787** [comment]
> > Could you remove AppVeyor as required check for PR submission in the setting?
> 
> Disabled AppVeyor webhook.

Thanks! Checks are now passing.

**Comment by kalyanac** [comment]
Can you mention in the description what is the oldest XCode and OSX versions this will support and what versions are getting dropped?

**Comment by wyk9787** [comment]
> Can you mention in the description what is the oldest XCode and OSX versions this will support and what versions are getting dropped?

Done.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
+![](https://github.com/google/pprof/workflows/ci/badge.svg)
```
Comment (aalexand) [inline README.md:1]: The badge links to the image itself which is not useful and is uncommon to do. Please make the badge link to the CI status page, like the AppVeyor badge did and like Travis CI and Codecov badges do.

Comment (wyk9787) [reply README.md:1]: Sent https://github.com/google/pprof/pull/549.

**Pull request merged**